packages/ui-pattern-block/src/components/CodeEditor.tsx

```diff
@@ -0,0 +1,54 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import { ReactElement, useEffect, useState } from 'react';
+import { SandpackCodeEditor, useActiveCode, useSandpack } from '@codesandbox/sandpack-react';
+import { Toolbar } from './Toolbar';
+import { SandpackTemplate } from '../types';
+
+interface Props {
+    template: SandpackTemplate;
+    showResetButton: boolean;
+    showSandboxLink: boolean;
+    showResponsivePreview: boolean;
+    shouldCollapseCodeByDefault: boolean;
+    onResponsivePreviewOpen: () => void;
+    onCodeChange: (filename: string, code: string) => void;
+    onResetToDefault: () => void;
+}
+
+export const CodeEditor = ({
+    onCodeChange,
+    onResetToDefault,
+    onResponsivePreviewOpen,
+    template,
+    showResetButton,
+    showResponsivePreview,
+    showSandboxLink,
+    shouldCollapseCodeByDefault,
+}: Props): ReactElement => {
+    const { code } = useActiveCode();
+    const { sandpack } = useSandpack();
+    const { activeFile } = sandpack;
+    const [isEditorCollapsed, setIsEditorCollapsed] = useState(shouldCollapseCodeByDefault);
+
+    useEffect(() => {
+        onCodeChange(activeFile, code);
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [code]);
```

Comment:
This one seems odd, as there is a bug and I can't adjust it. As the adjustment got stored and then another call will be made with the old code and overwrite the change.

Reply:
Fixed!! Thanks, great catch